lib/nxt_support/refinements/crystalizer.rb

```diff
@@ -0,0 +1,11 @@
+module NxtSupport
+  module Refinements
+    module Crystalizer
```

Comment:
Quick idea (might be not thought through): what about calling it `ArrayCrystalizer` to make it more clear that it refines `Array`?

Reply:
I just realised refining `Enumerable` works with Ruby 3.0. I'll keep the name and switch the behaviour based on ruby version.